freqtrade/plugins/pairlist/MarketCapPairList.py

```diff
@@ -0,0 +1,92 @@
+"""
+MarketCap PairList provider
+
+Provides dynamic pair list based on market cap
+"""
+import logging
+from typing import Any, Dict, List
+
+import requests
+from cachetools.ttl import TTLCache
+
+from freqtrade.exceptions import OperationalException
+from freqtrade.plugins.pairlist.IPairList import IPairList
+
+
+logger = logging.getLogger(__name__)
+
+
+class MarketCapPairList(IPairList):
+    def __init__(self, exchange, pairlistmanager,
+                 config: Dict[str, Any], pairlistconfig: Dict[str, Any],
+                 pairlist_pos: int) -> None:
+        super().__init__(exchange, pairlistmanager, config, pairlistconfig, pairlist_pos)
+
+        if 'number_assets' not in self._pairlistconfig:
+            raise OperationalException(
```

Comment:
I think you'll also want to check if `number_assets` is numeric / an integer - by providing a non-int entry, i'm pretty sure i can break the response from coingecko ...

Reply:
Check added.